File: features/discovery/home/src/main/java/com/tokopedia/home/beranda/presentation/view/adapter/itemdecoration/HomeRecyclerDecoration.java
package com.tokopedia.home.beranda.presentation.view.adapter.itemdecoration;

import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.view.View;

public class HomeRecyclerDecoration extends RecyclerView.ItemDecoration {
    private int spacing;

    public HomeRecyclerDecoration(int spacing) {
        this.spacing = spacing;
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        outRect.bottom = spacing;
    }
}

File: features/discovery/home/src/main/java/com/tokopedia/home/beranda/presentation/view/adapter/itemdecoration/HomeRecyclerDecoration.kt
package com.tokopedia.home.beranda.presentation.view.adapter.itemdecoration

import android.graphics.Rect
import androidx.recyclerview.widget.RecyclerView
import android.view.View

class HomeRecyclerDecoration(private val spacing: Int) : RecyclerView.ItemDecoration() {

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        val itemCount = parent.layoutManager!!.itemCount - 1
        val currentPosition = parent.layoutManager!!.getPosition(view)
        if (currentPosition in 3 until itemCount) {
            outRect.bottom = spacing
        }
    }
}
